src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/CaseNoteRepository.kt

```diff
@@ -1,7 +1,11 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository
 
-import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.repository.PagingAndSortingRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.CaseNote
 import java.util.UUID
 
-interface CaseNoteRepository : JpaRepository<CaseNote, UUID>
+interface CaseNoteRepository : PagingAndSortingRepository<CaseNote, UUID> {
+  fun findAllByReferralId(referralId: UUID, pageable: Pageable?): Page<CaseNote>
```

Comment:
what happens if `pageable` is null?

Reply:
if pageable is null then it returns all case notes (no pagination)